### PR TITLE
Setupheartbeat

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

**Comment by shriramshankar** [comment]
I noticed a few TODO comments here https://github.com/salesforce/refocus-collector/blob/master/src/config/config.js#L96 
Wondering if we want to keep those default values instead?

**Comment by shriramshankar** [comment]
This is ready. Since this is based on #111 and #112, let's merge them before getting to this.

File: src/heartbeat/heartbeat.js [hunk 1]
```diff
@@ -9,168 +9,53 @@
...
-
-    //return req;
-  }
-  catch (err) {
-    throw err;
+  const collectorName = config.name;
+  const refocusUrl = config.refocus.url;
+  const accessToken = config.refocus.accessToken;
+  const proxy = config.refocus.proxy;
+  const heartbeatEndpoint = `/v1/${collectorName}/heartbeat`;
```
Comment (jgraff2) [inline src/heartbeat/heartbeat.js]: Shouldn't it be /v1/collectors/${collectorName}/heartbeat?

File: src/heartbeat/heartbeat.js [hunk 2]
```diff
@@ -9,168 +9,53 @@
...
+  const urlToPost = refocusUrl + heartbeatEndpoint;
+
+  const existing = configModule.getConfig().metadata;
+  const current = u.getCurrentMetadata();
+  const changed = u.getChangedMetadata(existing, current);
+  Object.assign(existing, current);
+  const requestbody = {
+    logLines: [],
```
Comment (jgraff2) [inline src/heartbeat/heartbeat.js]: This should be set for real now that we're actually sending the heartbeat to refocus.

File: test/heartbeat/heartbeat.js [hunk 3]
```diff
@@ -58,303 +80,221 @@ const generator3 = {
...
+  ],
+  generatorsUpdated: [],
+  generatorsDeleted: [
+    generator1,
+    generator2,
+    generator3,
+  ],
+};
+
+const hbResponseWithSGToUPdate = {
```
Comment (jgraff2) [inline test/heartbeat/heartbeat.js]: Update

File: test/heartbeat/heartbeat.js [hunk 4]
```diff
@@ -58,303 +80,221 @@ const generator3 = {
...
       done();
-    });
+    }).catch((err) => done(err));
   });
 
-  it('sendHeartbeat - missing token', (done) => {
-    config.name = 'Test';
-    config.refocus.url = url;
-    delete config.refocus.collectorToken;
+  it('Ok, simple response without generators', (done) => {
```
Comment (jgraff2) [inline test/heartbeat/heartbeat.js]: duplicate tests.

File: src/heartbeat/heartbeat.js (hunk 2 above)
Comment (shriramshankar) [reply src/heartbeat/heartbeat.js]: The initial intent was to have the logs sent with the heartbeat but later we decided to have a buffered queue post to the v1/auditEvents endpoint directly. @iamigo Can you confirm this? I will delete this L43 once I get the confirmation.

Comment (iamigo) [reply src/heartbeat/heartbeat.js]: confirmed, shriram

**Pull request merged**